tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -0,0 +1,279 @@
+            _discoveryService = discoveryService;
+            _configurationPoller = configurationPoller;
+            _lineProbeResolver = lineProbeResolver;
+            _debuggerSink = debuggerSink;
+            _probeStatusPoller = probeStatusPoller;
+            _unboundProbes = new List<ProbeDefinition>();
+        }
+
+        public static LiveDebugger Instance => LazyInstance.Value;
+
+        private static LiveDebugger Create()
+        {
+            var source = GlobalSettings.CreateDefaultConfigurationSource();
+            var settings = ImmutableDebuggerSettings.Create(DebuggerSettings.FromSource(source));
+            if (!settings.Enabled)
+            {
+                return Create(settings, null, null, null, null, null);
+            }
+
+            var apiFactory = DebuggerTransportStrategy.Get(settings.AgentUri);
+            IDiscoveryService discoveryService = DiscoveryService.Create(source, apiFactory);
+
+            var probeConfigurationApi = ProbeConfigurationApiFactory.Create(settings, apiFactory, discoveryService);
+            var updater = ConfigurationUpdater.Create(settings);
+            IConfigurationPoller configurationPoller = ConfigurationPoller.Create(probeConfigurationApi, updater, settings);
+
+            var snapshotStatusSink = SnapshotSink.Create(settings);
+            var probeStatusSink = ProbeStatusSink.Create(settings);
+
+            var batchApi = BatchUploadApiFactory.Create(settings, apiFactory, discoveryService);
+            var batchUploader = BatchUploader.Create(batchApi);
+            var debuggerSink = DebuggerSink.Create(snapshotStatusSink, probeStatusSink, settings, batchUploader);
+
+            var lineProbeResolver = LineProbeResolver.Create();
+            var probeStatusPoller = ProbeStatusPoller.Create(settings, probeStatusSink);
+
+            return Create(settings, discoveryService, configurationPoller, lineProbeResolver, debuggerSink, probeStatusPoller);
+        }
+
+        // for tests
+        internal static LiveDebugger Create(ImmutableDebuggerSettings settings, IDiscoveryService discoveryService, IConfigurationPoller configurationPoller, ILineProbeResolver lineProbeResolver, IDebuggerSink debuggerSink, IProbeStatusPoller probeStatusPoller)
+        {
+            var debugger = new LiveDebugger(settings, discoveryService, configurationPoller, lineProbeResolver, debuggerSink, probeStatusPoller);
+            debugger.Initialize();
+
+            return debugger;
+        }
+
+        private void Initialize()
+        {
+            if (!_settings.Enabled)
+            {
+                Log.Information("Live Debugger is disabled. To enable it, please set DD_DEBUGGER_ENABLED environment variable to 'true'.");
+                return;
+            }
+
+            if (_settings.TransportType != TracesTransportType.Default)
+            {
+                Log.Information("Live Debugger is not supported on UDS or named pipelines.");
```

Comment:
Nit and can be done in a later PR.
```suggestion
                Log.Information("Live Debugger is not supported on UDS or named pipes.");
```

Reply:
I'm prettifying the history in a separate branch so I'll address this comment in a follow-up PR.

As to your second question regarding the impact on package size - great point. I verified the differences between the binaries that were built on Master and in this branch and here are the results:

On Windows, the Managed DLL is larger in ~140KB and the Native DLL is larger in ~180KB.
On Linux, the Managed DLL is larger in ~140KB and the Native SO is larger in ~380KB.